doc/spring-boot-plugin-dev.md

```diff
@@ -0,0 +1,318 @@
+# Spring Boot Plugin Development
+
+This document explores the use cases and methodology for implementing,
+packaging and integrating Spring Boot applications which are extensible by external plugins.
+This document refers to application and library code written in Kotlin and managed by Maven,
+but should be applicable to any other equivalent technologies.
```

Comment:
by equivalent do you mean spring supported? if so then it maybe reads better as "other spring supported technologies/languages"